packages/taquito-rpc/test/taquito-rpc.spec.ts

```diff
@@ -4381,6 +4396,212 @@ describe('RpcClient test', () => {
     });
   });
 
+  describe('smartRollupPublish', () => {
+    it('should have correct types to access smart_rollup_publish results', async (done) => {
+      httpBackend.createRequest.mockReturnValue(Promise.resolve(smartRollupPublishResponse));
+      const response = await client.getBlock();
+      const content = response.operations[0][0]
+        .contents[0] as OperationContentsAndResultSmartRollupPublish;
```

Comment:
Regarding the conversation we had about avoiding casting, we can modify this code to make sure that type narrowing works well for our types and users can avoid casting.

```
const content = response.operations[0][0].contents[0]; // no casting here
// test fields that are common here
if (content.kind !== OpKind.SMART_ROLLUP_PUBLISH) {
  fail('Some message');
  return; //maybe this is not needed.
}
// test fields that are only specific to OperationContentsAndResultSmartRollupPublish
```